src/parser/postgresparser.cpp

```diff
@@ -405,41 +405,65 @@ expression::AbstractExpression* PostgresParser::ConstTransform(A_Const* root) {
   return ValueTransform(root->val);
 }
 
-// This function takes in a Postgres FuncCall parsenode and transfers it into
-// a Peloton FunctionExpression object.
-// TODO: support function calls on a single column.
 expression::AbstractExpression* PostgresParser::FuncCallTransform(
     FuncCall* root) {
   expression::AbstractExpression* result = nullptr;
-  std::string type_string =
-      (reinterpret_cast<value*>(root->funcname->head->data.ptr_value))->val.str;
-
-  type_string = "AGGREGATE_" + type_string;
-
-  if (root->agg_star) {
-    expression::AbstractExpression* children = new expression::StarExpression();
-    result = new expression::AggregateExpression(
-        StringToExpressionType(type_string), false, children);
-  } else {
-    if (root->args->length < 2) {
-      // auto children_expr_list = TargetTransform(root->args);
-      expression::AbstractExpression* child;
-      auto expr_node = (Node*)root->args->head->data.ptr_value;
+  std::string fun_name =
+      StringUtil::Lower(
+          (reinterpret_cast<value*>(root->funcname->head->data.ptr_value))->val.str);
+
+  if (!IsAggregateFunction(fun_name)) {
+    // Normal functions (i.e. built-in functions or UDFs)
+    fun_name = (reinterpret_cast<value*>(root->funcname->tail->data.ptr_value))->val.str;
+    std::vector<expression::AbstractExpression*> children;
+    for (auto cell = root->args->head; cell != nullptr; cell = cell->next) {
+      auto expr_node = (Node*) cell->data.ptr_value;
       if (expr_node->type == T_A_Expr) {
-        child = AExprTransform((A_Expr*)expr_node);
+        children.push_back(AExprTransform(reinterpret_cast<A_Expr *>(expr_node)));
       } else if (expr_node->type == T_A_Const) {
-        child = ConstTransform((A_Const*)expr_node);
+        children.push_back(ConstTransform(reinterpret_cast<A_Const *>(expr_node)));
       } else if (expr_node->type == T_ColumnRef) {
-        child = ColumnRefTransform((ColumnRef*)expr_node);
-      } else {
-        throw NotImplementedException(
-          "Function within Aggregate is not supported yet\n");
+        children.push_back(ColumnRefTransform(reinterpret_cast<ColumnRef *>(expr_node)));
+      } else if (expr_node->type == T_FuncCall){
+        children.push_back(FuncCallTransform(reinterpret_cast<FuncCall*>(expr_node)));
       }
+      else {
+        std::ostringstream oss;
+        oss<<"Type "<<expr_node->type<<"is not supported in function call";
+        throw NotImplementedException(oss.str());
```

Comment:
I guess you can combine the above three lines into "throw NotImplementedException(StringUtil::Format("Type %d is not supported in function call\n", expr_node->type));"